corehq/apps/app_manager/xform.py

```diff
@@ -2129,3 +2148,36 @@ def infer_vellum_type(control, bind):
         })
         return None
     return result['name']
+
+
+def find_missing_instances(wrapped_xform):
+    from corehq.apps.app_manager.suite_xml.post_process.instances import get_all_instances_referenced_in_xpaths
+    instance_declarations = wrapped_xform.get_instance_ids()
+    missing_instances = set()
+    missing_unknown_instance = set()
+    instances, unknown_instance_ids = get_all_instances_referenced_in_xpaths('', [wrapped_xform.render()])
```

Comment:
Assuming rendering is relatively low cost, but wanted to point out that it could be avoided since [it's rendering the form anyway](https://github.com/dimagi/commcare-hq/blob/07d0faccb079371fae2eb8c645275f5149c82048/corehq/apps/app_manager/models.py#L995) it might make sense to add an `xform_xml` argument to `find_missing_instances` and `check_for_missing_instances`?